net/golioth/rpc.c

```diff
@@ -189,9 +189,6 @@ int golioth_rpc_register(struct golioth_client *client,
 	method->callback_arg = callback_arg;
 
 	client->rpc.num_methods++;
-	if (client->rpc.num_methods == 1) {
-		status = golioth_rpc_observe(client);
-	}
```

Comment:
If user will call `golioth_rpc_observe()` alone or after `golioth_rpc_register()`, then there is a chance of using `client->rpc_mutex` without first initializing it (as it happens in `golioth_rpc_register()`). Let's move this mutex initialization to be part of `golioth_init()` (but with some helper function like `golioth_rpc_init()`). Also, it would be good idea to move `struct k_mutex rpc_mutex;` into `struct golioth_rpc rpc;`.